### ISSUE TITLE
Getting flannel to tolerate custom taints

### ISSUE THREAD

**Issue by ashic** [kubernetes/kops#3873]
I have the following ig definition:

```
apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  creationTimestamp: 2017-06-29T17:49:13Z
  labels:
    kops.k8s.io/cluster: mycluster.com
  name: influxdb
spec:
  image: kope.io/k8s-1.7-debian-jessie-amd64-hvm-ebs-2017-07-28
  machineType: m4.xlarge
  maxSize: 1
  minSize: 1
  nodeLabels:
    app: influxdb
  role: Node
  rootVolumeSize: 500
  subnets:
  - eu-west-1a
  taints:
  - mything/dedicated=influxdb-monitoring:NoSchedule

```

I recently switched from weave to flannel by doing kops edit cluster -> setting network to flannel. This was needed as weave wasn't playing nice after the 1.7.10 upgrade from 1.6.2. However, the flannel pod isn't running on that node (or any nodes that have NoSchedule taints. Is there a way to automatically provide flannel the tolerations needed to run on these nodes? 

kops edit cluster shows:

```
...
  networking:
    flannel: {}
...
```
Perhaps the tolerations can be specified there? Are there any examples of this?

**Comment by justinsb** [comment]
So @ashic the flannel pods should have this toleration:

```
      tolerations:
      - operator: Exists
```

Which should mean they run on all tainted nodes.

So if you `kubectl get daemonset -n kube-system kube-flannel-ds` do you see that toleration value?

I'm wondering if the problem is actually something else (e.g. resource limits).  If you `kubectl describe ds  -n kube-system kube-flannel-ds` are there any events or errors?  Or if you `kubectl describe node <node>` for one of the nodes where it isn't running, what do the resources look like, and any events / errors there?

**Comment by chrislovecnm** [comment]
Bump - remove blocks next?

**Comment by iMartyn** [comment]
I can confirm that on 1.8.4 kube installed by kops 1.8.0-beta.1 (close to master but not quite) the tolerations on flannel as installed out of the box are : 
```yaml
      tolerations:
      - effect: NoSchedule
        key: node-role.kubernetes.io/master
```
Which allows for scheduling on masters on 1.8 and as I understand it, also 1.7.

I suspect that the 1.6 taints are still on the masters post-upgrade but that's more an upgrade from 1.6 to 1.7 issue than an issue that should block 1.8 of kops releasing?

**Comment by chrislovecnm** [comment]
No removing blocks next

**Comment by ashic** [comment]
@justinsb Sorry.... had resolved the issue, but forgot to get back. The missing toleration was the issue. I added it manually. The settings now show:

```
   key  :  node-role.kubernetes.io/master
   effect : NoSchedule

   operator :  Exists
   effect : NoSchedule
```

and it all works.

**Comment by justinsb** [comment]
I believe this is fixed, but only in the kops 1.8 branch.  The toleration I see is:

```
  tolerations:
  - operator: Exists
...
```

Please reopen / ping if this continues in a 1.8.0 (or the next beta if we end up doing one!)
